plugins/userprofileservice/services/rest_ups.go

```diff
@@ -0,0 +1,115 @@
+/****************************************************************************
+ * Copyright 2022, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+// Package services //
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/optimizely/agent/plugins/userprofileservice"
+	"github.com/optimizely/go-sdk/pkg/decision"
+	"github.com/optimizely/go-sdk/pkg/logging"
+	"github.com/optimizely/go-sdk/pkg/utils"
+	"github.com/rs/zerolog/log"
+)
+
+// RestUserProfileService represents the rest API implementation of UserProfileService interface
+type RestUserProfileService struct {
+	Requester  *utils.HTTPRequester
+	Host       string            `json:"host"`
+	Headers    map[string]string `json:"headers"`
+	LookupPath string            `json:"lookupPath"`
+	SavePath   string            `json:"savePath"`
+}
+
+// Lookup is used to retrieve past bucketing decisions for users
+func (r *RestUserProfileService) Lookup(userID string) (profile decision.UserProfile) {
+	if userID == "" {
+		return
```

Comment:
Is it ok not returning profile in go?

Reply:
yes, if we don't do it here, it will just return empty profile struct.